src/redisgears_python.c

```diff
@@ -1594,7 +1706,12 @@ static Record* RedisGearsPy_ToPyRecordMapperInternal(Record *record, void* arg){
     switch(RedisGears_RecordGetType(record)){
     case STRING_RECORD:
         str = RedisGears_StringRecordGet(record, &len);
+        // try to first decode it as string, if fails create a byte array.
```

Comment:
Isn't there some checktype method instead of relying on failure?

Reply:
On our side we can not know what the byte array represents. I am not sure if Python expose an api to check if bytearray is a valid string or not.